playbooks/upload_to_swift.yml

```diff
@@ -3,10 +3,42 @@
   connection: local
   gather_facts: False
   tasks:
-    - name: Create archive
-      command: "tar -cjf {{ archive_name }} {{ artifacts_dir | basename }}"
+
+    - set_fact:
+        archive_base_name: "{{job_name}}_{{build_number}}"
+
+    # This is done so that downloaded archives have a useful name
+    - name: Move artifacts dir
+      shell: mv "{{ artifacts_dir | basename }}" "{{ archive_base_name }}"
       args:
         chdir: "{{ artifacts_dir | dirname }}"
+
+    - set_fact:
+        adir: "{{ artifacts_dir | dirname }}/{{archive_base_name}}"
+
+    # links are unlikely to work when expanded on a different
+    # system so remove them. Create a list of removed links
+    # to give users a clue as to why their file is missing.
+    # Also swift client fails on dangling links.
+    - name: Remove links from artifacts
+      shell: |
+        find {{ adir | basename }} \
+          -type l \
+          -print \
+          -delete \
+          > {{ adir | basename }}/removedlinks.txt
+      args:
+        chdir: "{{ adir |dirname }}"
+
+    # This task runs async while the individual files are being uploaded
+    # then when compression is complete, the resultant archive is uploaded.
+    - name: Create archive
+      shell: "tar -c {{ adir | basename }} | gzip --fast > {{archive_base_name}}.tar.gz"
+      args:
+        chdir: "{{ adir | dirname }}"
+      async: 10800
```

Comment:
It'd be super-nice to have a comment telling the reader that this is a 3 hour timeout.

Reply:
Done